### DISCUSSION TITLE
Mapping the NAS to a Windows Machine

### DISCUSSION THREAD

**Discussion by Flip01002003** [alexhorner/WD-DX4000-Installer#11] [General]
Hey! I appreciate the assistance with the new version last month or so. You came in CLUTCH with the quick response and solution! You the man, for sure!

I was able to install the OS on a flash drive and can successfully tunnel to the DX4000 from my native system. I was successful from my Windows system and my Ubuntu VM. Now I'm just having issues actually mapping the device to my Windows system. I'm not new to Linux, but I'm also not an expert so it could VERY well be a user error lol.

Also, if it's important to mention, the DX4000 shuts down after about 10 or 15 minutes, even with active use. I do not have hard drives for it currently; I'm using an additional flash drive for testing and diagnosing purposes. Let me know if this has anything to do with it. 

Lastly, the user account I created on my install does is not in the sudoers group; is there a username/password I can use to have god-rights to my NAS?

Any input or assistance is appreciated!

**Comment by alexhorner** [comment]
The shutdown is very concerning... Are there any mentions of it in the `dmesg` logs? If not and it's an unclean shutdown I'd be concerned that there's some kind of thermal issue, maybe bad paste or fan not running fast enough/at all?

Root user is something you would have been asked for a password for during install IIRC. You can use this to add your normal user to sudoers.

Mapping? You mean accessing files? For basic access you can use SFTP with something like WinSCP. If you're talking about integration with Windows Explorer as a network drive, you'll need to find a Samba tutorial.

**Reply by Flip01002003** [reply]
Apologies for the late reply. Most of my issues were just trying to figure out how to set everything up. As far as root access, that was a durrr moment for me. I used su - to gain root access and added the account I created during installation to the sudoers group. Easy peasy.

 I think I'm on the right path at this point. I've installed Cockpit on the server and got the web application up and running. I'm working on creating a network share at the moment.

My biggest concern is this shutdown issue. Looking at the dmesg, OOM killer appears to be crashing my NAS. When I power it back on after it crashes, it comes back up almost instantaneously, which is interesting. I can SSH back into it almost immediately. If I do a graceful shutdown, it takes about 5 minutes to be able to SSH back into it.

It could very well be a hardware error; this thing had been sitting around untouched for quite some time.

**Reply by alexhorner** [reply]
@Flip01002003 Interesting... Next time it crashes, can you try pulling the plug before you turn it back on and see if you get the same results?

I am wondering if you installed a GUI during install, and along with it, all of the stuff that handles power management such as putting the machine to sleep at sensible intervals.... like 15 minutes! Additionally, as far as I remember, there is no visual indicator the machine is asleep as you would expect from a normal machine like a flashing light, merely as a result of the way the front panel indicators work on this machine being non-standard.

A GUI will also be wasting a lot of resources in the background I'd imagine, as it will be running on the VGA output of the unit constantly (yes it has VGA, no it isn't accessible by default but yes you can break it out with some work)

**Reply by Flip01002003** [reply]
It was doing this ever since I completed the install. I did monitor the memory on the Cockpit web GUI but it didn't really show me anything crazy happening (I don't know exactly how accurate it is though). I don't think I installed a GUI during the installation. I was SSHing into the thing up until I got Cockpit to work. 

I did not go through the stpes yet to get the LCD/fan working properly. Maybe it has something to do with that?? 

I plan on messing with it tomorrow. Between getting that to work, and doing some yard work, I'm beat lol. I'll definitely let you know what I can figure out!

**Reply by alexhorner** [reply]
LCD and fan won't affect it as the default fan is full speed. Can you check whether xorg is installed?

**Reply by Flip01002003** [reply]
It is installed and up to date.

**Reply by alexhorner** [reply]
It is? That means you do have some GUI stuff installed and I suspect that's causing your troubles

**Reply by Flip01002003** [reply]
You think I should straight uninstall xorg? Would that mess with Cockpit at all, you think?

**Reply by Flip01002003** [reply]
Update: I took a snapshot of the stats according to Cockpit. This is where I'm sitting at right as the system crashes. Yeah, memory is in low supply, but it stayed steady just under 1GB the entire time its running. 

![image](https://github.com/user-attachments/assets/fc4ca723-b074-4ea3-a5b5-c69fccce2392)

**Comment by Flip01002003** [comment]
journalctl -p3 -r gave me this:

![image](https://github.com/user-attachments/assets/ff7214db-7490-4b1d-bd56-807ef51d6458)

Lots of errors from systemd-cryptsetup. I wonder if this has something to do with it. It doesn't appear the CPU or RAM is actually being overloaded. But again, I'm not sure if the stats I see in Cockpit are REAL, REAL time stats. 

Any recommendations?

**Reply by alexhorner** [reply]
If I'm honest I'd start a fresh. xorg merely indicates you installed some kind of GUI during install but trying to remove a GUI from a system is, quite frankly, hell. The cryptsetup erroring suggests there's a misconfigured encrypted volume too. I'd reinstall and make sure you read all of the screens carefully, deselecting everything you don't need (GUI is selected by default) and go from there

**Reply by Flip01002003** [reply]
I'm in the process of reinstalling now. I'll let you know how it goes. 

As far as the encryption thing, I know for a fact I did not elect to use encryption, which is kind of weird. I did, however, leave the desktop environment and gnome options checked the first time by mistake. That could be causing my issues. 

This time around I'm only selecting the standard system utilities.

**Comment by Flip01002003** [comment]
I'm having a heck of a time getting this to work this time around for whatever reason. I get through the installation process with ease. When it comes to the step where I mount is where I'm having trouble. 

I know my installation drive (USB) is sda; I don't even see sdb until I plug in my boot drive (other USB). Once I do, the only drive I can mount is sdb2, which appears to have a file system. I do the follwoing:

disk-detect
mkdir /mnt/mountpoint 
mount /dev/sdb2 /mnt/mountpoint
**At this point, if I run ls mnt/mountpoint/ I get a file system with bin, boot, dev, etc, home, etc. The startup.nsh is already present.

When I reboot and remove the installation drive, I get nothing. The NAS doesn't appear in my nmap scan at all. It only shows up if I insert the installation drive again. 

I don't remember if there was something special I did last time, but it aint working for me this time :(

Any advice??

**Reply by Flip01002003** [reply]
Nevermind, I figured it out. I don't think it was anything I was doing; I had to reset a few times and nmap finally populated it with a different IP address. I probably should make this one a static IP, huh?

I'm installing Cockpit now and the NFS module. If there is any interest I can report back when it's all loaded and I have a share set up. Like I said, Im running a second thumb drive as a testbed for this project just to be sure it works. 

I haven't had any crashes as of yet and it's been solid running for about an hour now! So far so good!

**Reply by alexhorner** [reply]
Would certainly love to hear back, I like to hear when people have success with this project!

**Reply by alexhorner** [reply]
I should reiterate that a USB installation is not the best for longevity as I have found in my own running system the USBs wear out after maybe 8-12 months due to all the logging and general system writes that happen.

I have a system with the additional SATA port soldered to the motherboard which allows me to install an SSD as a 5th drive which tucks nicely in the side panel, accessible from the drive bays without getting in the way of disks.

https://community.wd.com/t/my-conversion-of-a-dx4000-to-ubuntu-server/245639/22

**Reply by Flip01002003** [reply]
Oh yeah the two USB thing is just a temporary thing u til i can actually get some hard drives for it! Mainly just trying to get the process down and understand the setup and niches with it all! Also to test the access from other devices in the house, including phones and TVs and stuff!

I did get Cockpit up and running which makes managing super easy and not super taxing on the hardware. I like having the web browser access.

I do like that additional SATA port idea! I may do that one down the road if all goes well! 

Do you know if there is anyway way to increase the RAM?? I haven’t taken it apart to see what the board looks like or anything so I’m not 100% sure if that’s possible or even worth the time/effort. It might help with stability though.

**Reply by alexhorner** [reply]
It just takes laptop style DDR3, I think mine has 4GB installed. The BIOS is **extremely** picky though and I haven't succeeded in finding a module it likes aside from the original, neither equal size or an upgrade.
